### PR TITLE
add overview doc

### PR DESCRIPTION
**Pull request by krancour**

### CODE REVIEW

**Review by kencochrane — APPROVED** [review]
LGTM

File: docs/docs/10-overview.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+environments, sources of materials (such as container images or Kubernetes
+manifests), _how_ to apply those materials to each environment (typically by
+interacting with repositories, configuration management tools, and Argo CD), and
+the conditions under which new materials may progress from one logical
+environment to the next.
+
+## Next steps
+
+To learn more about Kargo, consider checking out our
+[concepts doc](./concepts) or get hands-on right away with our
```
Comment (kencochrane) [inline docs/docs/10-overview.md:58]: this is failing due to a broken link here, since https://github.com/akuity/kargo/pull/328 needs to be merged first.

Comment (krancour) [reply docs/docs/10-overview.md:58]: That's correct.

**Comment by krancour** [comment]
Not waiting for full CI run for a docs-only change.

**Pull request merged**